src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala

```diff
@@ -127,6 +139,31 @@ object DepositHandler {
   }
 
   private def extractBag(mimeType: String)(implicit settings: Settings, id: String): Try[File] = {
+    def checkAvailableDiskspace(file: File): Try[Unit] = {
+      val zipFile = new ZipFile(file.getPath)
+      val headers = zipFile.getFileHeaders.asScala.asInstanceOf[JListWrapper[FileHeader]] // Look out! Not sure how robust this cast is!
+      val uncompressedSize = headers.map(_.getUncompressedSize).sum
+      val availableDiskSize = Files.getFileStore(file.toPath).getUsableSpace
+      log.debug(s"Available (usable) disk space currently $availableDiskSize bytes. Spaces needed: $uncompressedSize bytes. Margin required: ${settings.marginDiskSpace} bytes.")
+      if (uncompressedSize + settings.marginDiskSpace > availableDiskSize)
+        Failure(RejectedDepositException(id, "Not enough disk space to process deposit.",
+          new IllegalStateException(s"Required disk space for unzipping: ${uncompressedSize + settings.marginDiskSpace} (including ${settings.marginDiskSpace} margin). Available: $availableDiskSize")))
+      else Success(())
+    }
+
+    def checkDiskspaceForMerging(files: Seq[File]): Try[Unit] = {
+      val requiredSpace = files.map(_.length).sum
+      files.headOption.map {
+        f =>
+          val availableDiskSize = Files.getFileStore(f.toPath).getUsableSpace
```

Comment:
is this duplicated from the function above? I guess this makes up for another function of type `Path => Try[Unit]`?